### ISSUE TITLE
mvn versions:display-plugin-updates NullPointerException if using a plugin without specifying its version

### ISSUE THREAD

**Issue by dtreskunov** [mojohaus/versions-maven-plugin#430]
This is probably not high-priority since the usage is invalid, but the error message is really not helpful in this case.

```xml
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>test</groupId>
  <artifactId>test</artifactId>
  <version>test</version>
  <packaging>jar</packaging>

  <name>test</name>

  <build>
    <pluginManagement>
      <plugins>
        <!-- removing this results in NullPointerException at org.codehaus.mojo.versions.DisplayPluginUpdatesMojo.pad
        <plugin>
          <groupId>org.codehaus.mojo</groupId>
          <artifactId>license-maven-plugin</artifactId>
          <version>1.20</version>
        </plugin>
        -->
      </plugins>
    </pluginManagement>
  </build>

  <reporting>
    <plugins>
      <plugin>
        <groupId>org.codehaus.mojo</groupId>
        <artifactId>license-maven-plugin</artifactId>
      </plugin>
    </plugins>
  </reporting>
</project>
```

```
mvn versions:display-plugin-updates -e
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------------< test:test >------------------------------
[INFO] Building test test
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- versions-maven-plugin:2.8.1:display-plugin-updates (default-cli) @ test ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 19.052 s
[INFO] Finished at: 2020-10-05T13:44:22-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:versions-maven-plugin:2.8.1:display-plugin-updates (default-cli) on project test: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.8.1:display-plugin-updates failed.: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:versions-maven-plugin:2.8.1:display-plugin-updates (default-cli) on project test: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.8.1:display-plugin-updates failed.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.8.1:display-plugin-updates failed.
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.NullPointerException
    at org.codehaus.mojo.versions.DisplayPluginUpdatesMojo.pad (DisplayPluginUpdatesMojo.java:713)
    at org.codehaus.mojo.versions.DisplayPluginUpdatesMojo.execute (DisplayPluginUpdatesMojo.java:477)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

**Comment by pzygielo** [comment]
I can't reproduce. It works fine on my end (maven 3.6.3 + openjdk 1.8, 11; maven 2.2.1 + openjdk 1.8).

For 3.6.3 I get:
```
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------------< test:test >------------------------------
[INFO] Building test test
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- versions-maven-plugin:2.8.1:display-plugin-updates (default-cli) @ test ---
[INFO] 
[INFO] All plugins with a version specified are using the latest versions.
[INFO] 
[WARNING] The following plugins do not have their version specified:
[WARNING]   maven-clean-plugin ........................ (from super-pom) 2.2
[WARNING]   maven-compiler-plugin ................... (from super-pom) 2.0.2
[WARNING]   maven-deploy-plugin ....................... (from super-pom) 2.4
[WARNING]   maven-install-plugin ...................... (from super-pom) 2.2
[WARNING]   maven-jar-plugin .......................... (from super-pom) 2.1
[WARNING]   maven-resources-plugin .................... (from super-pom) 2.2
[WARNING]   maven-site-plugin .................. (from super-pom) 2.0-beta-4
[WARNING]   maven-surefire-plugin ................... (from super-pom) 2.4.2
[WARNING]   org.codehaus.mojo:license-maven-plugin  (from super-pom) 1.0-beta-2
[INFO] 
[WARNING] Project does not define minimum Maven version, default is: 2.0
[INFO] Plugins require minimum Maven version of: 3.5.4
[INFO] Note: the super-pom from Maven 3.6.3 defines some of the plugin
[INFO]       versions and may be influencing the plugins required minimum Maven
[INFO]       version.
[INFO] 
[ERROR] Project does not define required minimum version of Maven.
[ERROR] Update the pom.xml to contain maven-enforcer-plugin to
[ERROR] force the Maven version which is needed to build this project.
[ERROR] See https://maven.apache.org/enforcer/enforcer-rules/requireMavenVersion.html
[ERROR] Using the minimum version of Maven: 3.5.4
[INFO] 
[INFO] Require Maven 2.0.2 to use the following plugin updates:
[INFO]   maven-site-plugin ............................. 3.9.1 -> 2.0-beta-7
[INFO] 
[INFO] Require Maven 2.0.6 to use the following plugin updates:
[INFO]   maven-clean-plugin ................................... 3.1.0 -> 2.5
[INFO]   maven-deploy-plugin ............................. 3.0.0-M1 -> 2.8.1
[INFO]   maven-install-plugin ............................ 3.0.0-M1 -> 2.5.1
[INFO]   maven-jar-plugin ..................................... 3.2.0 -> 2.4
[INFO]   maven-resources-plugin ............................... 3.2.0 -> 2.6
[INFO]   maven-site-plugin .................................. 3.9.1 -> 2.0.1
[INFO]   maven-surefire-plugin ........................... 3.0.0-M5 -> 2.4.3
[INFO]   org.codehaus.mojo:license-maven-plugin ............... 2.0.0 -> 1.6
[INFO] 
[INFO] Require Maven 2.0.9 to use the following plugin updates:
[INFO]   maven-compiler-plugin ................................ 3.8.1 -> 3.1
[INFO]   maven-surefire-plugin ............................ 3.0.0-M5 -> 2.17
[INFO] 
[INFO] Require Maven 2.1.0 to use the following plugin updates:
[INFO]   maven-site-plugin .................................. 3.9.1 -> 2.1.1
[INFO] 
[INFO] Require Maven 2.2.0 to use the following plugin updates:
[INFO]   maven-site-plugin .................................... 3.9.1 -> 3.0
[INFO] 
[INFO] Require Maven 2.2.1 to use the following plugin updates:
[INFO]   maven-clean-plugin ................................. 3.1.0 -> 2.6.1
[INFO]   maven-compiler-plugin ................................ 3.8.1 -> 3.3
[INFO]   maven-deploy-plugin ............................. 3.0.0-M1 -> 2.8.2
[INFO]   maven-install-plugin ............................ 3.0.0-M1 -> 2.5.2
[INFO]   maven-jar-plugin ..................................... 3.2.0 -> 2.6
[INFO]   maven-resources-plugin ............................... 3.2.0 -> 2.7
[INFO]   maven-site-plugin .................................... 3.9.1 -> 3.4
[INFO]   maven-surefire-plugin ............................ 3.0.0-M5 -> 2.20
[INFO]   org.codehaus.mojo:license-maven-plugin ............... 2.0.0 -> 1.8
[INFO] 
[INFO] Require Maven 3.0 to use the following plugin updates:
[INFO]   maven-clean-plugin .......................................... 3.1.0
[INFO]   maven-jar-plugin ................................... 3.2.0 -> 3.1.0
[INFO]   maven-resources-plugin ............................. 3.2.0 -> 3.1.0
[INFO]   maven-site-plugin .................................. 3.9.1 -> 3.7.1
[INFO]   maven-surefire-plugin .......................... 3.0.0-M5 -> 2.22.0
[INFO]   org.codehaus.mojo:license-maven-plugin .............. 2.0.0 -> 1.17
[INFO] 
[INFO] Require Maven 3.0.4 to use the following plugin updates:
[INFO]   maven-compiler-plugin .............................. 3.8.1 -> 3.7.0
[INFO] 
[INFO] Require Maven 3.0.5 to use the following plugin updates:
[INFO]   maven-compiler-plugin ....................................... 3.8.1
[INFO]   maven-deploy-plugin ...................................... 3.0.0-M1
[INFO]   maven-install-plugin ..................................... 3.0.0-M1
[INFO]   maven-jar-plugin ............................................ 3.2.0
[INFO]   maven-site-plugin ........................................... 3.9.1
[INFO]   maven-surefire-plugin .................................... 3.0.0-M5
[INFO]   org.codehaus.mojo:license-maven-plugin .............. 2.0.0 -> 1.20
[INFO] 
[INFO] Require Maven 3.1.0 to use the following plugin updates:
[INFO]   maven-resources-plugin ...................................... 3.2.0
[INFO] 
[INFO] Require Maven 3.5.4 to use the following plugin updates:
[INFO]   org.codehaus.mojo:license-maven-plugin ...................... 2.0.0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.876 s
[INFO] Finished at: 2020-10-06T10:05:18+02:00
[INFO] ------------------------------------------------------------------------
```
and very similar for maven 2.2.1:
```
[INFO] Scanning for projects...
[INFO] Searching repository for plugin with prefix: 'versions'.
[INFO] Ignoring available plugin update: 2.8.1 as it requires Maven version 3.0
[INFO] Ignoring available plugin update: 2.7 as it requires Maven version 3.0
[INFO] Ignoring available plugin update: 2.6 as it requires Maven version 3.0
[INFO] Ignoring available plugin update: 2.5 as it requires Maven version 3.0
[INFO] Ignoring available plugin update: 2.4 as it requires Maven version 3.0
[INFO] ------------------------------------------------------------------------
[INFO] Building test
[INFO]    task-segment: [versions:display-plugin-updates]
[INFO] ------------------------------------------------------------------------
[INFO] [versions:display-plugin-updates {execution: default-cli}]
[INFO] 
...
[INFO] 
[INFO] Require Maven 3.5.4 to use the following plugin updates:
[INFO]   org.codehaus.mojo:license-maven-plugin ........................ 2.0.0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESSFUL
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 51 seconds
[INFO] Finished at: Tue Oct 06 10:07:21 CEST 2020
[INFO] Final Memory: 61M/374M
```

**Comment by khmarbaise** [comment]
I can reproduce this issue by using an older Maven versions: 3.0.5, 3.1.1, 3.2.5, 3.3.1, 3.3.9, 3.5.2 (JDK8) starting with 3.6.0, 3.6.1, 3.6.2 and 3.6.3 it works. 

So it's not an issue of versions-maven-plugin.
